velox/exec/Task.cpp

```diff
@@ -418,6 +420,51 @@ void Task::removeSpillDirectoryIfExists() {
   }
 }
 
+/// Builds the query trace config.
+std::optional<QueryTraceConfig> Task::maybeMakeTraceConfig() const {
+  const auto& queryConfig = queryCtx_->queryConfig();
+  if (!queryConfig.queryTraceEnabled()) {
+    return std::nullopt;
+  }
+
+  const auto queryTraceDir = queryConfig.queryTraceDir();
+  VELOX_CHECK(
+      !queryTraceDir.empty(),
+      "Query trace enabled but the trace dir is not set properly");
+
+  const auto queryTraceNodes = queryConfig.queryTraceNodes();
+  if (queryTraceNodes.empty()) {
+    LOG(WARNING) << "Query trace enabled but no target nodes.";
```

Comment:
```
s/but no target nodes./but no tracing plan node specified/
```
Shall we also fail this query as empty trace directory? We will only use this feature for debugging purpose.